src/qt/bitcoingui.cpp

```diff
@@ -82,10 +82,10 @@ BitcoinGUI::BitcoinGUI(interfaces::Node& node, const PlatformStyle *_platformSty
     m_network_style(networkStyle)
 {
     QSettings settings;
-    if (!restoreGeometry(settings.value("MainWindowGeometry").toByteArray())) {
-        // Restore failed (perhaps missing setting), center the window
-        move(QGuiApplication::primaryScreen()->availableGeometry().center() - frameGeometry().center());
-    }
+    settings.beginGroup("MainWindow");
+    resize(settings.value("size", QSize(800, 540)).toSize());
```

Comment:
Not sure hard-coding a size in pixels is a good idea. It should auto-size based on what Qt determines is needed when a wallet is loaded (ie, all widgets visible)

Reply:
This is a default size that is used only at first run or after `-resetguisettings`.

Reply:
I'm aware. But the ideal default size isn't measured in pixels.

Reply:
> But the ideal default size isn't measured in pixels.

Correct. The ideal size is set by users and it will be saved in settings. I don't think the default size could be ideal for more than one person :)

Reply:
Until we removed the default wallet, the default window size was automatically sane for any platform. We just need to go back to that.

Reply:
@luke-jr Do you suggest to revert https://github.com/bitcoin/bitcoin/pull/15454 back or to calculate the default window size as if a wallet is loaded?

Reply:
The latter of course

Reply:
Reworked in #116.